drizzle/resample.py

```diff
@@ -554,6 +642,32 @@ def add_image(self, data, exptime, pixmap, scale=1.0,
                 out_ctx=None,
             )
 
+        if data2 is None:
+            ninputs2 = None
+        else:
+            if isinstance(data2, np.ndarray):
+                ninputs2 = 1
+                if data2.shape != data.shape:
+                    raise ValueError(
+                        "'data2' shape is not consistent with 'data' shape."
+                    )
+            else:
+                shapes2 = set()
+                ninputs2 = len(data2)
+                data2 = list(data2)
+                for k, d in enumerate(data2):
+                    if d is None or d.size == 0:
+                        data2[k] = None
+                    else:
+                        shapes2.add(d.shape)
+
+                if len(shapes2) > 1 and shapes2.pop() != data.shape:
+                    raise ValueError(
+                        "'data2' shape(s) is not consistent with 'data' shape."
+                    )
```

Comment:
> Why set data2 to None here? 

Do you mean `data2[k] = None` on line 660? If so, that's being done when the error/variance array are empty. Setting to `None` allows code to not resample that variance array.

> And it looks like you're only checking one of the data shapes in shapes2 against data.shape - is that what was intended?

`shapes2` is a set and all input arrays should have all the same shape. Appending the same `ndarray.shape` to a set will result in a set with only one element. Yes, it is needed because JWST pipeline has unit tests that try to resample variance arrays of different shapes from the `data` array.

Reply:
Oh, okay, to handle when d.size == 0?  That makes sense.

For shapes2, what I meant to ask was - what if data2 is a list with arrays of different shapes, so the set has more than one element, but the first one popped matches the input data?  Then it won't raise the ValueError.  Maybe you meant 'or' instead of 'and'?

Reply:
> Maybe you meant 'or' instead of 'and'?

Yes, this line needs to be fixed